### PR TITLE
Add MonadLogger instace for SqlQueryT

### PR DESCRIPTION
**Pull request by ikv**

### CODE REVIEW

**Comment by brandonchinn178** [comment]
I would rather not add a `monad-logger` dependency in this package.

You can always do
```hs
newtype MyMonad a = MyMonad (SqlQueryT (ReaderT ...) a)

instance MonadLogger MyMonad where
  monadLoggerLog a b c d = MyMonad $ lift $ monadLoggerLog a b c d
```

or just define an orphan instance in your own code
```hs
instance MonadLogger m => MonadLogger (SqlQueryT m) where
  monadLoggerLog a b c d = lift $ monadLoggerLog a b c d
```

The `MonadReader` instance added in #40 was fine because we already had a dependency on `mtl`, and the `exceptions` instances added in #39 were fine because exceptions are common enough (and manually implementing `MonadMask` is a bit annoying)

**Comment by ikv** [comment]
But `persistent` depends on, and uses extensively `monad-logger`. Why you do not want to add direct dependency?

**Comment by brandonchinn178** [comment]
Oh right I forgot `persistent` depends on `monad-logger`, although I wouldn't say it uses it extensively. It's pretty much only used by the functions that initialize a SQL connection. But sure, it doesn't add any transitive dependencies, so this is fine

File: package.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ library:
     - conduit >= 1.3.0.3 && < 1.4
     - containers >= 0.5.10.2 && < 0.7
     - exceptions >=0.8 && <0.11
+    - monad-logger ^>= 0.3.29
```
Comment (brandonchinn178) [inline package.yaml]:
```suggestion
    - monad-logger >= 0.3.29 && < 0.4
```

**Review by brandonchinn178 — APPROVED** [review]
one small change

**Pull request merged**